cache.go

```diff
@@ -87,8 +89,11 @@ type InMemoryCache[K comparable, V any] struct {
 	cleanupCh   chan struct{}
 	closeCh     chan struct{}
 	closeOnce   sync.Once
-	closed      int32     // Atomic flag indicating cache is closed
-	itemPool    sync.Pool // Object pool for cache items
+	closed      int32     // Flag indicating cache is closed
+	itemPool    sync.Pool // Object pool
+
+	cachedTime   int64         // Cached timestamp updated by background goroutine
```

Comment:
Did you consider using [atomic.Int64](https://pkg.go.dev/sync/atomic#Int64)

These types were added with Go 1.19 to simplify the usage of atomic